src/proxy/sessions/session.rs

```diff
@@ -134,8 +136,11 @@ impl SessionArgs {
 impl Session {
     /// internal constructor for a Session from SessionArgs
     async fn new(args: SessionArgs) -> Result<Self> {
-        let addr = (std::net::Ipv4Addr::UNSPECIFIED, 0);
-        let socket = Arc::new(UdpSocket::bind(addr).await.map_err(Error::BindUdpSocket)?);
+        let socket = Arc::new(
```

Comment:
Won't this fail if you end up with more than one session (or endpoint)? Since each session will attempt to bind to the same port, and will then fail, and 

I'm not quite sure this will work 🤔 Maybe each endpoint you could be able to specify the port?

The proxy is built to be non-transparent, so Sessions do use unique sender and receiver information as it's storage for individual Sessions:

https://github.com/googleforgames/quilkin/blob/786d0d3561ad758aee7aa90966f9c106e4847f30/src/proxy/sessions/session_manager.rs#L26-L27

Breaking that contract could have some unintended consequences as well.

I would also advocate that there is some integration tests for this to ensure it works as expected.

Reply:
Just reviewing @XAMPPRocky 's [original comment ](https://github.com/googleforgames/quilkin/issues/345#issuecomment-887297745
)

Rather than bind a new port on each session, we probably need to do something more like having an `Option<Arc<UdpSocker>>` that gets passed into the Session, and if it's `None` then we bind to the ephemeral port.

Still not quite sure how/if that solves the unique key issue within `SessionManager` though 🤔 I'll write some thoughts on the original PR for some design discussion.

Reply:
Thanks for review @markmandel, good catch. Tested on live Quilkin process but with one session. That should de definitely covered in integration test. Looking forward to design discussions, meanwhile will pause dev work on this.